packages/xsnap/scripts/get_xsnap_version.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+set -ueo pipefail
+
+# the xsnap binary lives in a platform-specific directory
+unameOut="$(uname -s)"
+case "${unameOut}" in
+  Linux*) platform=lin ;;
+  Darwin*) platform=mac ;;
+  *) platform=win ;;
+esac
+
+# extract the xsnap package version from the long version printed by xsnap-worker
+./xsnap-native/xsnap/build/bin/${platform}/release/xsnap-worker -v | sed -e 's/^xsnap\s\+\([^\s]\+\)\s\+(XS\s\+[^)]\+)$/\1/g'
```

Comment:
This uses `sed` features that aren't available under Darwin (`\s` and `\+`), which makes me suspect they're GNUish or otherwise non-POSIX (possibly just a newer POSIX than macOS).  Can we use something more straightforward?

Also, please habitually quote shell variable expansions in things you'd expect to be a single token.  (They're also not strictly necessary for `vname=$anything` or `case $vname`.  Yes, shell quoting rules are really weird and have lots of corner cases.)

```suggestion
"./xsnap-native/xsnap/build/bin/${platform}/release/xsnap-worker" -v | sed -e 's/^xsnap  *\([^( ]*\).*/\1/'
```

Reply:
yeah, I got an error under linux (a docker container, under a macOS host):
```
node $ yarn build
yarn run v1.22.22
$ yarn build:bin && yarn build:env
$ if test -d ./test; then node src/build.js; else yarn build:from-env; fi
./scripts/get_xsnap_version.sh: 3: set: Illegal option -o pipefail
sh: 1: [[: not found
```

If I change the script to use `/bin/bash` instead of `/bin/sh`, it stops complaining about `-o pipefail`, but I still get the `[[: not found` error.

In both cases, the exit code is zero, so somehow this `[[: not found` is not actually causing a failure, so in my environment `yarn check-version` isn't really enforcing the version. I double-checked this by modifying my `package.json` to tweak the `version:` property, then re-ran `yarn check-version` without first doing a `yarn build`, and the exit code was still zero. So I think the `[[: not found` error is inhibiting the intended check and giving a false positive result.

If I change the package.json script line from `[[` to `[` (and `]]` to `]`), it seems to work as intended. Is the double-`[[` something to deal with quoting?

I'd suggest moving the comparison logic into a separate `scripts/check_xsnap_version.js`, which knows how to run xsnap as a child process, use a real `RegExp` to parse the result (avoiding `sed` and the tricky shell quoting), can sense whether `$npm_package_version` is missing (indicating someone is mistakenly running the script outside of a `yarn` context), and which can `exit(1)` internally. Basically, avoid using a shell for anything except launching a JS program :).

Reply:
`sed` is indeed a pain, it doesn't support common regular expression features unless you get into non posix land. Anyway, I simplified the expression. It should still work without issues as long as we don't tweak the `-v` output.